nir/src/main/scala/scala/scalanative/nir/Vals.scala

```diff
@@ -4,121 +4,174 @@ package nir
 import java.lang.Float.floatToRawIntBits
 import java.lang.Double.doubleToRawLongBits
 
+/** A NIR value. */
 sealed abstract class Val {
+
+  /** The type of the value. */
   final def ty: Type = this match {
-    case Val.Null                 => Type.Null
-    case Val.Zero(ty)             => ty
-    case Val.True | Val.False     => Type.Bool
-    case Val.Size(_)              => Type.Size
-    case Val.Char(_)              => Type.Char
-    case Val.Byte(_)              => Type.Byte
-    case Val.Short(_)             => Type.Short
-    case Val.Int(_)               => Type.Int
-    case Val.Long(_)              => Type.Long
-    case Val.Float(_)             => Type.Float
-    case Val.Double(_)            => Type.Double
-    case Val.StructValue(vals)    => Type.StructValue(vals.map(_.ty))
-    case Val.ArrayValue(ty, vals) => Type.ArrayValue(ty, vals.length)
-    case v: Val.ByteString        => Type.ArrayValue(Type.Byte, v.byteCount)
-    case Val.Local(_, ty)         => ty
-    case Val.Global(_, ty)        => ty
-
-    case Val.Unit     => Type.Unit
-    case Val.Const(_) => Type.Ptr
+    case Val.Null =>
```

Comment:
Is this what `scalafmt` does?

Reply:
scalamft adds inconsistent alignment when there are spaces in the case list. Further, aligning cases creates unnecessary changes in commits every time something changes in one specific case handler.

Reply:
Case in point: https://github.com/scala-native/scala-native/blob/f4b9078c6c96dade366518f037d8f47048d2a4f3/nir/src/main/scala/scala/scalanative/nir/Ops.scala#L11-L48

This function looks messy and scary to maintain. Having a more uniform way to format the match-expression that is resilient to changes is desirable, IMO. Tab-alignments of case handlers can look nice sometimes but they are also very high maintenance.

Reply:
Personally I'm more into the more compact style when most of match cases are single-liners - for me it seems way more readable and organised, however I understand the point about maintenance. I'm fine with both approaches, especially since this part of code is rarely changed.